sql/api/src/main/scala/org/apache/spark/sql/types/StringType.scala

```diff
@@ -61,6 +61,5 @@ class StringType private(val collationId: Int) extends AtomicType with Serializa
  */
 @Stable
 case object StringType extends StringType(0) {
-  val DEFAULT_COLLATION_ID = 0
   def apply(collationId: Int): StringType = new StringType(collationId)
```

Comment:
@dbatomic public API should take collation name instead of collation id.

Reply:
Ack. We will follow up on this.